spec/osut_tests_spec.rb

```diff
@@ -984,177 +975,243 @@ module M
-    expect(mod1.plenum?(nil, loops, setpoints)).to be(false)
+    expect(mod1.clean!).to eq(DBG)
+    expect(mod1.setpoints(nil)[:heating].nil?).to be(true)
     expect(mod1.debug?).to be(true)
     expect(mod1.logs.size).to eq(1)
-    expect(mod1.logs.first[:message]).to eq(m1)
+    expect(mod1.logs.first[:message]).to eq(ms3)
 
+    # --- --- --- --- --- --- --- --- --- --- --- --- --- --- --- --- --- --- #
+    file   = File.join(__dir__, "files/osms/in/seb.osm")
+    path   = OpenStudio::Path.new(file)
+    model  = translator.loadModel(path)
+    expect(model.empty?).to be(false)
+    model  = model.get
+    plenum = model.getSpaceByName("Level 0 Ceiling Plenum")
+    expect(plenum.empty?).to be(false)
+    plenum = plenum.get
+    expect(mod1.clean!).to eq(DBG)
 
-    if v < 350 # 5ZoneNoHVAC holds 1x OS:Material:AirWall, deprecated > 3.4.0
-      file = File.join(__dir__, "files/osms/in/5ZoneNoHVAC.osm")
-      path = OpenStudio::Path.new(file)
-      model = translator.loadModel(path)
-      expect(model.empty?).to be(false)
-      model = model.get
-      expect(mod1.clean!).to eq(DBG)
-      loops = mod1.airLoopsHVAC?(model)
-      expect(loops).to be(false)
-      expect(mod1.status.zero?).to be(true)
-      setpoints = mod1.heatingTemperatureSetpoints?(model)
-      expect(setpoints).to be(true)
-      expect(mod1.status.zero?).to be(true)
+    model.getSpaces.each do |space|
+      next if space == plenum
 
-      model.getSpaces.each do |space|
-        expect(mod1.plenum?(space, loops, setpoints)).to be(false)
-      end
+      zone = space.thermalZone
+      expect(zone.empty?).to be(false)
+      zone = zone.get
+      heat = mod1.maxHeatScheduledSetpoint(zone)
+      cool = mod1.minCoolScheduledSetpoint(zone)
 
-      expect(mod1.status.zero?).to be(true)
+      expect(heat[:spt]).to be_within(TOL).of(22.11)
+      expect(cool[:spt]).to be_within(TOL).of(22.78)
+      expect(heat[:dual]).to be(true)
+      expect(cool[:dual]).to be(true)
+
+      expect(space.partofTotalFloorArea).to be(true)
+      expect(mod1.plenum?(space)).to be(false)
+      expect(mod1.unconditioned?(space)).to be(false)
+      expect(mod1.setpoints(space)[:heating]).to be_within(TOL).of(22.11)
+      expect(mod1.setpoints(space)[:cooling]).to be_within(TOL).of(22.78)
     end
 
+    zone = plenum.thermalZone
+    expect(zone.empty?).to be(false)
+    zone = zone.get
+    heat = mod1.maxHeatScheduledSetpoint(zone) # simply returns model info
+    cool = mod1.minCoolScheduledSetpoint(zone) # simply returns model info
+
+    expect(heat[:spt].nil?).to be(true)
+    expect(cool[:spt].nil?).to be(true)
+    expect(heat[:dual]).to be(false)
+    expect(cool[:dual]).to be(false)
+
+    # "Plenum" spaceType triggers an INDIRECTLYCONDITIONED status; returns
+    # defaulted setpoint temperatures.
+    expect(plenum.partofTotalFloorArea).to be(false)
+    expect(mod1.plenum?(plenum)).to be(true)
+    expect(mod1.unconditioned?(plenum)).to be(false)
```

Comment:
Retrieving (valid) heating/cooling setpoints for a plenum, despite not have defined setpoints of its own.